src/fides/api/alembic/migrations/versions/1e750936a141_migrate_to_fideslang_2_0_uses_and_.py

```diff
@@ -0,0 +1,241 @@
+data_category_upgrades: Dict[str, str] = {
+    "user.credentials": "user.authorization.credentials",  # Verified in 2.0
+    "user.observed": "user.behavior",  # Verified in 2.0
+    "user.browsing_history": "user.behavior.browsing_history",  # Verified in 2.0
+    "user.media_consumption": "user.behavior.media_consumption",  # Verified in 2.0
+    "user.search_history": "user.behavior.search_history",  # Verified in 2.0
+    "user.organization": "user.contact.organization",  # Verified in 2.0
+    "user.non_specific_age": "user.demographic.age_range",  # Verified in 2.0
+    "user.date_of_birth": "user.demographic.date_of_birth",  # Verified in 2.0
+    "user.gender": "user.demographic.gender",  # Verified in 2.0
+    "user.political_opinion": "user.demographic.political_opinion",  # Verified in 2.0
+    "user.profiling": "user.demographic.profile",  # Verified in 2.0
+    "user.race": "user.demographic.race_ethnicity",  # Verified in 2.0
+    "user.religious_belief": "user.demographic.religious_belief",  # Verified in 2.0
+    "user.sexual_orientation": "user.demographic.sexual_orientation",  # Verified in 2.0
+    "user.financial.account_number": "user.financial.bank_account",  # Verified in 2.0
+    "user.genetic": "user.health_and_medical.genetic",  # Verified in 2.0
+}
+data_category_downgrades: Dict[str, str] = {
+    value: key for key, value in data_use_upgrades.items()
+}
+
+# Data Category fides keys are referenced in the following places:
+# - Datasets (fields, collections and dataset-level)
+# - Policy Rule - DONE
+# - Privacy Declaration - DONE
+# - Data Flow
+
+
+def update_datasets_data_categories(
+    bind: Connection, data_label_map: Dict[str, str]
+) -> None:
+    """Upgrade the datasets in the database to use the new data categories."""
+
+    # Get all datasets out of the database
+    existing_datasets: ResultProxy = bind.execute(
+        text("SELECT id, data_categories, collections FROM ctl_datasets;")
+    )
+
+    for row in existing_datasets:
+        # Update data categories at the top level
+        labels: List[str] = row["data_categories"]
+        updated_labels: List[str] = [
+            data_label_map.get(label, label) for label in labels
+        ]
+
+        update_label_query: TextClause = text(
+            "UPDATE datasets SET data_categories = :updated_labels WHERE id= :dataset_id"
+        )
+        bind.execute(
+            update_label_query,
+            {"dataset_id": row["id"], "updated_labels": updated_labels},
+        )
+
+        # Update the collections objects
+        collections = row["collections"]
+
+        for key, value in data_label_map.items():
+            collections.replace(key, value)
+
+        update_collections_query: TextClause = text(
+            "UPDATE datasets SET collections = :updated_labels WHERE id= :dataset_id"
+        )
+        bind.execute(
+            update_collections_query,
+            {"dataset_id": row["id"], "updated_labels": collections},
```

Comment:
nit: `updated_labels` feels like a weird name for this var? if just a c+p for now and you're in the process of cleaning up, please ignore me